Kyle-Law/private-events
# Merging Feature branch to development

In this project, I built a site similar to a private Eventbrite which allows users to create events and then manage user signups.

A user can create events. A user can attend many events. An event can be attended by many users. I modeled many-to-many relationships and also conscious of foreign keys and class names. Scopes are utilized to show upcoming/past events.

Click [here](https://kyle-private-events.herokuapp.com/) for the live link :)

Project snapshot:
![image](https://user-images.githubusercontent.com/55923773/80931634-d051aa80-8ded-11ea-9a8c-7dac9641bae5.png)

### Testing
1) Clone the project and cd to private-events
2) Run `bundle install` to install gems in Gemfile
3) Run `yarn install --check-files` to check and update yarn packages
4) Run 'rspec'
5) 17 examples, 0 failures should be shown. 

### Main Features
1) Without Signing In, You can't see 'Add New Event' link.
2) Click `Sign up` for new user, or `Sign In` for returning user.
3) Once signing in, the user can create a new event or see his profile.
4) The user session will be deleted upon clicking `Sign Out`.
5) You can Register / Unregister events on the event#show page.
6) You can see attending/attended events in user profile.
7) You can see attendees in event#show page.
8) You can check for upcoming/past events. 

Thanks. Looking forward to any suggestions and reviews :)

@Kyle-Law is permitted to work alone throughout the Rails session by student success due to a lack of available coding partner in UTC+1.
